dg-core/src/main/java/org/finra/datagenerator/distributor/ProcessingStrategy.java

```diff
@@ -0,0 +1,40 @@
+/*
+ * Copyright 2014 DataGenerator Contributors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.finra.datagenerator.distributor;
+
+import org.finra.datagenerator.consumer.DataPipe;
+
+import java.io.IOException;
+import java.util.Map;
+
+/**
+ * Created by Brijesh on 9/29/2015.
+ */
+public interface ProcessingStrategy {
+
+    /**
+     * Takes the map generated from SCXMLFrontier and pass it
+     * to the consumer for data processing
+     *
+     * @param map map of String and String
+     * @throws IOException io Exception
+     * @return DataPipe
+     *
+     */
+    DataPipe processOutput(Map<String, String> map) throws IOException;
```

Comment:
I am processing data by passing it to DataConsumer and then calling transformer. I need to return processed data through Data Pipe.